zproject/backends.py

```diff
@@ -1392,6 +1396,13 @@ class GitLabAuthBackend(SocialAuthMixin, GitLabOAuth2):
     # we just use the primary email address, which is always verified.
     # (No code is required to do so, as that's the default behavior).
 
+@external_auth_method
+class AppleAuthBackend(SocialAuthMixin, AppleIdAuth):
+    sort_order = 200
+    name = "apple-id"
+    auth_backend_name = "Apple"
+    display_icon = "/static/images/landing-page/logos/apple-icon.png"
```

Comment:
One other comment:

Does this produce a button that follows the guidelines Apple provides specifically for "Sign in with Apple" buttons? See here:
https://developer.apple.com/design/human-interface-guidelines/sign-in-with-apple/overview/buttons/

There is provision there for making a custom button. But there are a number of details if so.

Reply:
Alternatively, Apple provides on a CDN complete buttons in a range of styles:
https://developer.apple.com/documentation/signinwithapplejs/incorporating_sign_in_with_apple_into_other_platforms

Note in particular the `locale` parameter, for controlling the language of the text.

Reply:
> Does this produce a button that follows the guidelines Apple provides specifically for "Sign in with Apple" buttons?

Didn't know much about `locale` parameter, So thought we'll have problem with translations.
will try to using CDN complete buttons provided by apple.

Reply:
The buttons provided by apple have the logo centre aligned meaning it was just infront of "Sign in with Apple" text. If we use that it makes that look very odd from other "Log in with X" buttons. May be should create a custom one with their guidelines. If we are doing it we might want to adjust other buttons also( like changing "Log in" to "Sign in", maintaining the ratio of font-size to button size suggested by apple), so this would blend in the page well like the other buttons.

Reply:
Hmm, I see.

I'd like to avoid blocking this on any broader design changes to our auth buttons. So for getting this merged, what I'd like to prioritize is
* Have the Apple button compatible with Apple's guidelines.
* Keep the work simple, to focus on the rest of it and getting it to something we can merge and deploy.

I think it's OK to push to a followup, for after this is merged, the task of:
* Make the design of the Apple auth button in line with our other auth buttons.

How's that sound?

For that set of priorities, I think the simplest way will most likely be to use a button from the CDN, and accept that the logo is placed differently from the other buttons' logos. But a custom button following the guidelines would also be good if that's easier.

Reply:
Okay, for now, we'll use the button from CDN.